Lib/pydoc.py

```diff
@@ -2544,9 +2544,38 @@ def bltinlink(name):
             'key = %s' % key, '#ffffff', '#ee77aa', '<br>'.join(results))
         return 'Search Results', contents
 
+    def validate_source_path(path):
+        for importer, modname, ispkg in pkgutil.walk_packages():
+            try:
+                spec = pkgutil._get_spec(importer, modname)
+            except SyntaxError:
+                # raised by tests for bad coding cookies or BOM
+                continue
+            loader = spec.loader
+            if hasattr(loader, 'get_source'):
+                try:
+                    source = loader.get_source(modname)
+                except Exception:
+                    continue
+                if hasattr(loader, 'get_filename'):
+                    sourcepath = loader.get_filename(modname)
+                    if path == sourcepath:
+                        return
+            else:
+                try:
+                    module = importlib._bootstrap._load(spec)
+                except ImportError:
+                    continue
+                sourcepath = getattr(module, '__file__', None)
+                if path == sourcepath:
+                    return
+        else:
+            raise ValueError('not found {found}')
```

Comment:
Maybe this code can be factorized with ModuleScanner.run()?

Reply:
Unfortunately it cannot. ModuleScanner.run() contains too much specific to its primary function: it only calls callback if key is not None and is found in module name or description.

But it is a temporary solution. For 3.10 we can change API (pass not path, but a module name, so the loop will be unnecessary).